client/js/requests.js

```diff
@@ -0,0 +1,38 @@
+async function getItem(id) {
+  try {
+    const resp = await fetch(`http://localhost:3000/posts/${id}`);
+    const data = await resp.json();
+    return data;
+  } catch (err) {
+    console.warn(err);
+  }
+}
+
+async function postPost(e) {
+  e.preventDefault();
+  try {
+    const formDataObject = Object.fromEntries(new FormData(e.target));
+    if (formDataObject.hasOwnProperty("image")) {
+      formDataObject.resources = {};
+      formDataObject.resources.image = formDataObject.image;
+      delete formDataObject.image;
+    }
+    const formData = JSON.stringify(formDataObject);
+
+    const options = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: formData
+    };
+
+    const resp = await fetch("http://localhost:3000/posts", options);
+    const { id, err } = await resp.json();
```

Comment:
I could be wrong bcs I haven't tested this but I don't think this will work. You are destructing 'err' but the api returns 'error'